src/wp-includes/class-wp-block-patterns-registry.php

```diff
@@ -165,14 +165,16 @@ public function unregister( $pattern_name ) {
 	private function prepare_content( $pattern, $hooked_blocks ) {
 		$content = $pattern['content'];
 
-		$before_block_visitor = '_inject_theme_attribute_in_template_part_block';
+		$before_block_visitor = ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ? '_inject_theme_attribute_in_template_part_block' : null;
```

Comment:
We might consider moving the code that injects the `theme` attr to the patterns controller instead. Probably okay to leave as-is for now, but might be a nice follow-up post-6.4.

(Noting that if we still insert hooked blocks here, we'd be running `traverse_and_serialize_blocks` twice, which is suboptimal; so we'd also need to move hooked blocks insertion to the controller so we can remove it here; which would then require us to make sure it's run on the frontend. We can probably achieve that through running it in the Pattern block's `render_block`. This means that we'd need to make quite a few changes, which we most likely don't want to do before 6.4.)

Reply:
Can you explain why `traverse_and_serialize_blocks` would get called twice?

Reply:
> Can you explain why `traverse_and_serialize_blocks` would get called twice?

Assuming that we inject the `theme` attr in the patterns controller, but keep hooked blocks insertion here in the patterns registry (since it's needed both for the frontend and editor (i.e. REST API)) -- both of these call `traverse_and_serialize_blocks` 😅

Reply:
I see what you mean here - If we were to run the `theme` attribute injection in the REST API controller. Yes, good point that it would get called twice with different processing 👍🏻